nvtabular/dask/io.py

```diff
@@ -211,6 +216,10 @@ def __init__(
     def to_ddf(self, columns=None):
         return self.engine.to_ddf(columns=columns)
 
+    def to_iter(self):
```

Comment:
Im visualizing the `to_iter` method to actually create and return the iterator within this method (that is, do not greate an iterator attribute in `__init__`, but do it and return it right here.

Reply:
In the long run, the `Workflow` class will need to know when to convert the `Dataset` object into an iterator (`to_itr`) or a dask_cudf `DataFrame` (`to_ddf`).

Reply:
I suggest we accept a `columns` argument here (just like `to_itr`) and just return `self.engine.to_itr(columns=columns)`.   That is, we should probably do the real logic on the engine level.
 The engine can use `names=columns` when creating the iterator (and the engine type will already be known at that level). Overall, I would be hesitant to add any new arguments to the `DaskDataset` class here, most information should already be availbale.

Reply:
Summary of my previous comment:  Lets define a `to_itr` method on the `DatasetEngine` class and avoid passing itr specific arguments to `DaskDataset` - If anything, `GPUDatasetIterator` should be modified to accept the `part_size` (and/or `row_groups_per_part`) that we are already calculating here.